packages/dvilela/skills/memeooorr_abci/rounds.py

```diff
@@ -593,13 +607,51 @@ class ActionTweetRound(EventRoundBase):
     # Event.DONE, Event.NO_MAJORITY, Event.ROUND_TIMEOUT, Event.ERROR, Event.MISSING_TWEET
 
 
-class CheckFundsRound(EventRoundBase):
+class CheckFundsRound(CollectSameUntilThresholdRound):
     """CheckFundsRound"""
 
     payload_class = CheckFundsPayload  # type: ignore
+    synchronized_data_class = SynchronizedData
     extended_requirements = ()
+
     # This needs to be mentioned for static checkers
-    # Event.DONE, Event.NO_MAJORITY, Event.ROUND_TIMEOUT, Event.NO_FUNDS
+    # Event.DONE, Event.NO_MAJORITY, Event.ROUND_TIMEOUT,Event.NO_FUNDS, Event.SKIP
+
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
+        """Process the end of the block."""
+        if self.threshold_reached:
+            self.context.logger.info(
+                f"CheckFundsRound threshold reached: {self.threshold_reached}"
+            )
+            payload = CheckFundsPayload(
+                *(("dummy_sender",) + self.most_voted_payload_values)
+            )
+
+            if payload.check_funds_count >= 3 and payload.event == Event.NO_FUNDS.value:
+                self.context.logger.info(
+                    "check_funds_count >= 3 and event == Event.NO_FUNDS , skipping and moving to ResetRound"
+                )
+                return self.synchronized_data, Event.SKIP
+
+            if payload.check_funds_count < 3 and payload.event == Event.NO_FUNDS.value:
```

Comment:
For each of these paths, please add a short comment so we remember what each one of them is about